### DISCUSSION TITLE
getting revert in testFulFillRandomWordsPicksAWinnerResetsAndSendsMoney

### DISCUSSION THREAD

**Discussion by khushbusandhu** [Cyfrin/foundry-full-course-cu#646] [General]
why we did this in test 
```
  assert(previousTimeStamp < raffle.getLastTimeStamp());
```

**Comment by theirrationalone** [comment]
You might forgot to do.... `vm.deal(USER, 1ether);` at the end in the `setUp`

that second arg 1 ether can be any eligible eth amount in wei. 

1 ether is also interpolating into wei amount which is equals to 1e18 Or 1_00000000_0000000000 without _

**Reply by theirrationalone** [reply]
And see this line

```
[10859] RaffleTest::testRaffleRevertsWhenYouDontPayEnough()
```

You're actually Getting error in `testRaffleRevertsWhenYouDontPayEnough()` function where you might forgot to do `vm.expectRevet`

**Comment by theirrationalone** [comment]
Related to this`assert(previousTimeStamp < raffle.getLastTimeStamp());`

Have you noticed we did time travel via `vm.warp` because on mainnet we will have to wait for keepers to do things like announcing winner, and we setted up an interval of 30 seconds. So we're now have to simulate this behavior on local temporary anvil .... That's why we did `vm.warp`

previousTimeStamp is the timestamp when participators entered into lottery. 

Then we waited for 30 seconds.... 

And after announcing winner... Timepassed forwarded to 30 secs and `raffle.getLastTimeStamp());` gives current timestamp which is the latest timestamp also. So infact previousTimeStamp should be smaller than the current timestamp Or latest timestamp.

**Reply by khushbusandhu** [reply]
okay okay got it!
thanks @theirrationalone 

wouldn't be it irritating to you to clear these silly doubts of me 😢

**Reply by theirrationalone** [reply]
It's a revision for me, that's why i came here for, and Helping others just give me peace 😉 and that's not silly things, in starting I suffered maybe more than you. At that time I hadn't any idea about github & communities support on a specific course.

**Reply by khushbusandhu** [reply]
okay! so I can assume with more practice I will also get able to understand things with more clarity and I can also get able to resolve these "silly mistakes"? because right now I am thinking like should I continue this or just leave it here because i am not able to get things clearly in one go. i have to spend a lot of time in understanding even small points.

**Reply by theirrationalone** [reply]
Dont't leave it, In future you will be Thankful to it to Patrick(SIR)... It's a revolution btw. We all faced those problems but some of us took action against those things like you... And one more thing doubts are emerging in your head which is a Great sign, i'm not counting headache though 😂😅

**DON'T LEAVE IT**

**Reply by khushbusandhu** [reply]
okay, I will continue it till last at the cost of all my headaches😅
